### ISSUE TITLE
Inconsistent evaluation of editorconfig naming rule for inherited async methods

### ISSUE THREAD

**Issue by mareklinka** [dotnet/roslyn#34833]
**Version Used**: 
Visual Studio 2019 RTM
.NET Core 2.2.202

**Steps to Reproduce**:

1. Create new .NET Core console app project
2. Add an .editorconfig file to the solution
3. Add a naming rule for methods that says async methods should have the `Async` suffix
4. Create an interface with a Task-returning method without the `Async` suffix
5. Implement the interface in a class and make it async

**Expected Behavior**: VS reports the method it violating the naming rule
**Actual Behavior**: No violation is reported

I'm not 100% sure this is the right place to report this so if it isn't, I'll gladly go report it somewhere better.

At any rate, in the scenario above, the rule works perfectly well for non-inherited methods but fails for interface implementation.

The rule:
```
# Async methods should have "Async" suffix
dotnet_naming_rule.async_methods_end_in_async.symbols = any_async_methods
dotnet_naming_rule.async_methods_end_in_async.style = end_in_async
dotnet_naming_rule.async_methods_end_in_async.severity = error

dotnet_naming_symbols.any_async_methods.applicable_kinds = method
dotnet_naming_symbols.any_async_methods.applicable_accessibilities = *
dotnet_naming_symbols.any_async_methods.required_modifiers = async

dotnet_naming_style.end_in_async.required_prefix = 
dotnet_naming_style.end_in_async.required_suffix = Async
dotnet_naming_style.end_in_async.capitalization = pascal_case
dotnet_naming_style.end_in_async.word_separator =
```

Repro code:
``` csharp
public class Repro : IRepro
{
    public async Task DoWork()
    {
        await Task.Delay(1000);
    }

    public async Task DoOtherWork()
    {
        await Task.Delay(1000);
    }
}

public interface IRepro
{
    Task DoWork();
}
```

The result:
![image](https://user-images.githubusercontent.com/7479226/55739696-d5e19300-5a29-11e9-880b-a64e3a5d2061.png)

I'm also attaching a small repro solution:
[EditorConfigRepro.zip](https://github.com/dotnet/roslyn/files/3055270/EditorConfigRepro.zip)

**Comment by sharwell** [comment]
The following methods are excluded from naming rule analysis:

* Methods that implicitly implement an interface method (if the name is wrong, the interface should be updated)
* Methods that explicitly implement an interface method (the name is defined in the interface and the implementation cannot change it)
* Methods that override a method from a base class (the name is defined in the base class and the override cannot change it)

**Comment by mareklinka** [comment]
Thanks for a quick answer. So that would mean it's not actually possible to define this rule for such methods - because interface-defined methods cannot be marked as `async`. I think this quite limiting, especially for stuff like ASP.NET Core apps making extensive use of interfaces for DI and similar use cases. Is this something that might be changed eventually or was this a deliberate design decision that is set in stone for now?

**Comment by sharwell** [comment]
It's not really possible to specify the Async suffix rule in the naming conventions feature built into Visual Studio. We rely on the analyzers in [Microsoft/vs-threading](https://github.com/Microsoft/vs-threading) for checking this rule:
https://github.com/Microsoft/vs-threading/blob/master/doc/analyzers/VSTHRD200.md

**Comment by mareklinka** [comment]
Ah, that's too bad but thanks for the alternative solution, I'll give it a try!
